### ISSUE TITLE
UnsupportedType: Unsupported Redshift type: d

### ISSUE THREAD

**Issue by chtorr** [awslabs/aws-data-wrangler#48]
Hello!

I'm getting this error when trying to write a dataframe to Redshift while passing cast_columns.  I believe the source of the issue is related to this line https://github.com/awslabs/aws-data-wrangler/blob/master/awswrangler/pandas.py#L893 passing a dict to `data_types.convert_schema`, which then iterates like a list of tuples instead of dict key value pairs here  https://github.com/awslabs/aws-data-wrangler/blob/master/awswrangler/data_types.py#L294

Here's a trivial example:

```python
import awswrangler
import pandas as pd

df = pd.DataFrame({"id": [1, 2, 3], "name": ["name1", "name2", "name3"]})

session = awswrangler.Session()

schema = {
    "id": "BIGINT",
    "name": "VARCHAR"
}

session.pandas.to_redshift(
    df,
    's3://example-bucket/',
    None,
    "public",
    "example",
    "arn:aws:iam::123456789:role/rolename",
    cast_columns=schema,
)
```

Thanks for the great library!

**Comment by igorborgest** [comment]
Hey @chtorr, thank you very much for open this issue and for the precise troubleshooting!
With your analysis was easy fix the bug, the problem was exactly where you have pointed.

To fix I've just converted the received Dict to a List of Tuples expected by `data_types.convert_schema`.

In addition to that, I've also create a new test case for this situation and added types and [Mypy](https://github.com/python/mypy) coverage for this function.

**Comment by igorborgest** [comment]
This fix will be released in the version 0.0.11 later tonight. 🚀
